pandas/core/indexes/base.py

```diff
@@ -965,6 +965,10 @@ def take(
         taken = algos.take(
             self._values, indices, allow_fill=allow_fill, fill_value=self._na_value
         )
+
+        if not isinstance(taken, np.ndarray):
+            raise TypeError(f"Expected indexer to be of type np.ndarray instead received {type(taken)}")
```

Comment:
Another option is to check that `indices` (after the call to `ensure_platform_int` a few lines above, which will ensure the output is a numpy ndarray) is a 1D array. That would also catch the case where the user is passing a 2D indexer (which also gives strange results).

Reply:
Should I keep both the checks (for 1D array and type np.ndarray) or only implement the check for dimensions?